### ISSUE TITLE
First Level Headings Considered as List 

### ISSUE THREAD

**Issue by gokberkgunes** [vimwiki/vimwiki#1291]
If any number of spaces exists between a leading pound sign and a heading text, the text is considered as a list item for syntax highlighting. I believe this issue started happening after the commit 990b25ce37bbeb8a8c8f9208c0f8f455202041db. Additionally, this issue only happens if syntax is set to markdown; there is no issue if vimwiki's syntax is not changed to markdown.

I am able to reproduce the issue with a somewhat clean settings which has no other other plugins installed. I am using Neovim v0.8.1. Appending two screenshot to show the issue better way.

Latest commit:
![image](https://user-images.githubusercontent.com/96139011/208203769-7110f4f9-fbd2-4218-bfa1-934c4cb5c010.png)

When plugin set before specified commit:
![image](https://user-images.githubusercontent.com/96139011/208203488-58339aa4-6457-4154-a0c4-9c538fa62bd0.png)

**Comment by brennen** [comment]
Thanks for the report!

Oddly, I can't reproduce this under a mainline Vim build. With:

```
Version: 2022.12.02
Os: Linux
Vim: 801
Branch: dev
Revision: fea8bee
Date: 2022-12-04 21:25:52 -0700
```

I checked the following text:

```markdown
- one item
- two item
# three item

- list 1
* list 2
#   test
## test
```

Neither header is highlighted as a list item, as far as I can tell. For good measure I installed the following function:

```vim
    " hit ,S to debug current syntax highlighting groups
    " https://vim.fandom.com/wiki/Identify_the_syntax_highlighting_group_used_at_the_cursor
    "
    map <Leader>S :echo "hi<" . synIDattr(synID(line("."),col("."),1),"name") . '> trans<'
    \ . synIDattr(synID(line("."),col("."),0),"name") . "> lo<"
    \ . synIDattr(synIDtrans(synID(line("."),col("."),1)),"name") . ">"<CR>
```

And then hit `,S` with the cursor on those lines.  I get:

- `hi<VimwikiHeaderChar> trans<VimwikiHeaderChar> lo<Normal>` for the `#`
- `hi<VimwikiHeader1> trans<VimwikiHeader1> lo<VimwikiHeader1>` with the cursor on the `t` in `three item`

I don't see any difference in behavior with #1292 checked out, but I feel like I'm missing something. I'll have to see about getting a working Neovim going to test. I wouldn't be surprised if there's a Neovim-specific issue, I'd just like to have a better understanding before merging that change.

**Comment by lmburns** [comment]
I have this same issue in Neovim, but only whenever the file that I am editing is not found within the wiki that I have specified with `g.vimwiki_list`

**Comment by mkeeter** [comment]
I'm _also_ seeing this in Neovim, but haven't managed to isolate it yet.  Unlike @lmburns, I'm seeing it even when editing files within my `vimwiki_list` folder.  I'm configuring the plugin with [lazy.nvim](https://github.com/folke/lazy.nvim) and the following settings:
```lua
  {
    'vimwiki/vimwiki',
    config = function()
      vim.api.nvim_set_var('vimwiki_list', {
          {path = '~/wiki',
           syntax = 'markdown',
           ext = '.md' }})
      vim.cmd[[
        hi link VimwikiHeader1 pandocBlockQuoteLeader4
        hi link VimwikiHeader2 pandocBlockQuoteLeader3
        hi link VimwikiHeader3 pandocBlockQuoteLeader2
        hi link VimwikiHeader4 pandocBlockQuoteLeader1
        hi link VimwikiHeader5 pandocBlockQuoteLeader5
        hi link VimwikiHeader6 pandocBlockQuoteLeader6
        map <leader>w<leader>d :VimwikiDiaryPrevDay<CR>
        map <leader>w<leader>f :VimwikiDiaryNextDay<CR>
      ]]
    end
  },
```

**Comment by tinmarino** [comment]
I could reproduce with nvim, not with vim.

As @gokberkgunes spotted, this is due to some conflict with the list.

I created PR #1313  that has smaller change than PR #1292. The drawback of PR #1292 is that the markdown specific syntax file is repeating some command from the syntax generic one. I struggled to remove many duplications as those are hard to maintain and errors get easy to add ad hard to find when modifying one and not the other.

Can someone with nvim check #1313 please? @mkeeter @lmburns @gokberkgunes

**Comment by mkeeter** [comment]
Confirmed, this fixes the issue for me 🎉

**Comment by tinmarino** [comment]
Fixed by #1313. Thank you guys.
